### PR TITLE
Add Hermes Inspector Information

### LINKED ISSUE #439: Move Hermes Inspector Instructions to Docs Site
Currently on the main repo, but useful for a public audience (along with current Hermes docs in the docs site). https://github.com/microsoft/react-native-windows/blob/fedcbf60574ae67ec8a30f3f0c015f4070ee2f26/docs/inspector.md

### PR DESCRIPTION
**Pull request by chiaramooney**
### Why
Currently this information is on the main repo, but it would be useful for a public audience.

Resolves #439 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows-samples/pull/610)

### CODE REVIEW

File: docs/hermes.md [hunk 1]
```diff
@@ -62,3 +62,49 @@ pod 'libevent', :podspec => '../node_modules/react-native-macos/third-party-pods
...
+1. Click on the "Memory" tab in the inspector
+2. Heap snapshots and instrumented allocations should be working.
+
+### Enable debugging/profiling on release builds
+
+We keep the inspector turned off on release builds by default. If you want to debug or profile release builds, set the MSBuild property `EnableHermesInspectorInReleaseFlavor` to `'true'` when building the platform.
+
+### Known Issues
+
+1. CPU Sampling profiler current don't work
```
Comment (jonthysell) [inline docs/hermes.md:110]:
```suggestion
1. CPU Sampling profiler currently doesn't work

```

File: docs/hermes.md [hunk 2]
```diff
@@ -62,3 +62,49 @@ pod 'libevent', :podspec => '../node_modules/react-native-macos/third-party-pods
...
+
+### Steps to enable heap profiling
+
+Follow steps 1-7 from above, and then
+1. Click on the "Memory" tab in the inspector
+2. Heap snapshots and instrumented allocations should be working.
+
+### Enable debugging/profiling on release builds
+
+We keep the inspector turned off on release builds by default. If you want to debug or profile release builds, set the MSBuild property `EnableHermesInspectorInReleaseFlavor` to `'true'` when building the platform.
```
Comment (jonthysell) [inline docs/hermes.md:106]: Do we want to have an example on _how_ to do this? Like maybe the params to pass in run-windows, or how to add it to ExperimentalFeatures.props?

File: docs/hermes.md [hunk 3]
```diff
@@ -62,3 +62,49 @@ pod 'libevent', :podspec => '../node_modules/react-native-macos/third-party-pods
...
+5. Enable **Discover network targets** and **configure** the target discovery settings to include `localhost:8081` (or wherever the metro server is running)
+6. Within a few seconds "Hermes React Native" should appear on the page as a remote target
+7. Click on the **inspect** link to launch the `devtools` page
+8. Click `Ctrl+P` to open source files and set break points
+9. Alternatively, you can insert `debugger` statements in source code to break on specific locations
+
+In order to break on locations during boot, you can either
+
+- `debugger` statements during boot sequence will pause the runtime waiting for debugger to connect
+- Set break point and refresh the bundle through the Dev Server. The runtime will wait for debugger to attach.
```
Comment (jonthysell) [inline docs/hermes.md:96]:
```suggestion
- Set a break point and refresh the bundle through the Dev Server. The runtime will wait for debugger to attach.

```

File: docs/hermes.md [hunk 4]
```diff
@@ -62,3 +62,49 @@ pod 'libevent', :podspec => '../node_modules/react-native-macos/third-party-pods
...
+4. Navigate to `edge://inspect` in Edge browser or `chrome://inspect` in Chrome browser
+5. Enable **Discover network targets** and **configure** the target discovery settings to include `localhost:8081` (or wherever the metro server is running)
+6. Within a few seconds "Hermes React Native" should appear on the page as a remote target
+7. Click on the **inspect** link to launch the `devtools` page
+8. Click `Ctrl+P` to open source files and set break points
+9. Alternatively, you can insert `debugger` statements in source code to break on specific locations
+
+In order to break on locations during boot, you can either
+
+- `debugger` statements during boot sequence will pause the runtime waiting for debugger to connect
```
Comment (jonthysell) [inline docs/hermes.md:95]:
```suggestion
- Add `debugger` statements into the boot sequence to pause the runtime waiting for debugger to connect

```

**Review by jonthysell — APPROVED** [review]

File: docs/hermes.md (hunk 1 above)
Comment (asklar) [reply docs/hermes.md:110]:
```suggestion
1. The CPU Sampling profiler doesn't work
```

File: docs/hermes.md (hunk 4 above)
Comment (asklar) [reply docs/hermes.md:95]: consider making this a code snippet to make it clear that 
```js
  debugger;
``` 
here is literally what to write to break into the debugger.

File: docs/hermes.md [hunk 5]
```diff
@@ -62,3 +62,49 @@ pod 'libevent', :podspec => '../node_modules/react-native-macos/third-party-pods
 3. Run `pod install`
 
 > Be sure to set your target's deployment target to at least 10.14 before running `pod install`
+
+## Hermes Inspector
+
+*React Native for Windows* with *Hermes engine* now supports direct JavaScript runtime inspection using tools such as Chrome or Edge `devtools`, `VSCode` debugger, Flipper etc. by implementing an in-process Chrome Debug Protocol server.
```
Comment (asklar) [inline docs/hermes.md:68]:
```suggestion
*React Native for Windows* with *Hermes engine* supports direct JavaScript runtime inspection using tools such as Chrome or [Edge `devtools`](https://docs.microsoft.com/microsoft-edge/devtools-guide-chromium/), [`VSCode` debugger](https://code.visualstudio.com/Docs/editor/debugging), [Flipper](https://fbflipper.com/) etc. by implementing an in-process Chrome Debug Protocol server. 
```

**Pull request merged**